File: julia/eu001.jl
#!/usr/bin/env julia
using Test

# multiple determines if n is a multiple of any of the bases.
function multiple(n::Int, bases::Vector{Int})
    for b in bases
        if n%b == 0
            return true
        end
    end
    return false
end

# natmult computes the sum of all multiples of the given bases that are
# below n.
function natmult(n::Int, bases::Vector{Int})
    result = 0
    for i=1:(n-1)
        if multiple(i,bases)
            result += i
        end
    end
    result
end

bases = [3,5]

@test natmult(10,bases) == 23

# problem
println(natmult(1000,bases))

File: rust/src/eu001.rs
fn natmult(bases: &[i32], max: i32) -> i32 {
    let mut result = 0;
    for i in 0..max {
        for b in bases {
            if i % b == 0 {
                result += i;
                break;
            }
        }
    }
    println!("bases: {:?}; max: {}; result: {}",bases, max, result);
    result
}

fn main() {
    let bases = [3,5];
    let max = 10;
    if natmult(&bases[..],max) != 23 {
        panic!("test failed! expecting result: 23");
    }
    natmult(&bases[..],1000);
}
